In [12]:
import numpy as np
import pandas as pd


In [13]:
humidity = pd.Series([
    83, 80, 75, 68, 62, 60,
    58, 60, 66, 73, 79, 84
])

print(humidity)


0     83
1     80
2     75
3     68
4     62
5     60
6     58
7     60
8     66
9     73
10    79
11    84
dtype: int64


In [14]:
# Завдання 1

print(humidity.describe())


count    12.000000
mean     70.666667
std       9.547330
min      58.000000
25%      61.500000
50%      70.500000
75%      79.250000
max      84.000000
dtype: float64


In [15]:
# Завдання 2

range_value = humidity.max() - humidity.min()

variance_sample = humidity.var(ddof=1)
variance_population = humidity.var(ddof=0)

std_value = humidity.std(ddof=1)

q1 = humidity.quantile(0.25)
q3 = humidity.quantile(0.75)

iqr = q3 - q1

cv = std_value / humidity.mean() * 100

print("Розмах:", range_value)
print("Дисперсія ddof=1:", variance_sample)
print("Дисперсія ddof=0:", variance_population)
print("Стандартне відхилення:", std_value)
print("Q1:", q1)
print("Q3:", q3)
print("IQR:", iqr)
print("Коефіцієнт варіації:", cv, "%")


Розмах: 26
Дисперсія ddof=1: 91.15151515151516
Дисперсія ddof=0: 83.55555555555556
Стандартне відхилення: 9.547330263037681
Q1: 61.5
Q3: 79.25
IQR: 17.75
Коефіцієнт варіації: 13.510373013732565 %


In [16]:
print("Q1 з describe:", humidity.describe()["25%"])
print("Q1 вручну:", q1)

print("Q3 з describe:", humidity.describe()["75%"])
print("Q3 вручну:", q3)


Q1 з describe: 61.5
Q1 вручну: 61.5
Q3 з describe: 79.25
Q3 вручну: 79.25


In [17]:
# Завдання 3

df = pd.DataFrame({
    "Місто": [
        "Київ", "Львів", "Одеса", "Харків", "Дніпро",
        "Чернігів", "Івано-Франківськ", "Полтава", "Суми", "Ужгород"
    ],

    "Регіон": [
        "Центр", "Захід", "Південь", "Схід", "Схід",
        "Центр", "Захід", "Центр", "Схід", "Захід"
    ],

    "Січ": [84, 85, 82, 83, 82, 85, 86, 84, 85, 84],
    "Лют": [82, 83, 80, 80, 79, 83, 84, 81, 82, 82],
    "Бер": [78, 80, 78, 75, 74, 79, 81, 76, 77, 79],
    "Кві": [70, 73, 74, 68, 67, 71, 75, 69, 70, 73],
    "Тра": [65, 68, 71, 62, 61, 66, 70, 63, 64, 68],
    "Чер": [63, 65, 69, 60, 59, 63, 67, 61, 62, 66],
    "Лип": [61, 63, 67, 58, 57, 61, 65, 59, 60, 64],
    "Сер": [63, 65, 68, 60, 59, 63, 67, 61, 62, 66],
    "Вер": [68, 70, 72, 66, 65, 69, 72, 67, 68, 71],
    "Жов": [75, 77, 76, 73, 72, 76, 78, 74, 75, 77],
    "Лис": [81, 82, 80, 79, 78, 82, 83, 80, 81, 82],
    "Гру": [85, 86, 83, 84, 83, 86, 87, 85, 86, 85]
})

months = [
    "Січ", "Лют", "Бер", "Кві", "Тра", "Чер",
    "Лип", "Сер", "Вер", "Жов", "Лис", "Гру"
]

df["Середня_річна_вологість"] = df[months].mean(axis=1)

print(df)


              Місто   Регіон  Січ  Лют  Бер  Кві  Тра  Чер  Лип  Сер  Вер  \
0              Київ    Центр   84   82   78   70   65   63   61   63   68   
1             Львів    Захід   85   83   80   73   68   65   63   65   70   
2             Одеса  Південь   82   80   78   74   71   69   67   68   72   
3            Харків     Схід   83   80   75   68   62   60   58   60   66   
4            Дніпро     Схід   82   79   74   67   61   59   57   59   65   
5          Чернігів    Центр   85   83   79   71   66   63   61   63   69   
6  Івано-Франківськ    Захід   86   84   81   75   70   67   65   67   72   
7           Полтава    Центр   84   81   76   69   63   61   59   61   67   
8              Суми     Схід   85   82   77   70   64   62   60   62   68   
9           Ужгород    Захід   84   82   79   73   68   66   64   66   71   

   Жов  Лис  Гру  Середня_річна_вологість  
0   75   81   85                72.916667  
1   77   82   86                74.750000  
2   76   80   83    

In [18]:
east = df[df["Регіон"] == "Схід"]

print("Регіон Схід:")
print(east)

print("\nОписова статистика:")
print(east["Середня_річна_вологість"].describe())


Регіон Схід:
    Місто Регіон  Січ  Лют  Бер  Кві  Тра  Чер  Лип  Сер  Вер  Жов  Лис  Гру  \
3  Харків   Схід   83   80   75   68   62   60   58   60   66   73   79   84   
4  Дніпро   Схід   82   79   74   67   61   59   57   59   65   72   78   83   
8    Суми   Схід   85   82   77   70   64   62   60   62   68   75   81   86   

   Середня_річна_вологість  
3                70.666667  
4                69.666667  
8                72.666667  

Описова статистика:
count     3.000000
mean     71.000000
std       1.527525
min      69.666667
25%      70.166667
50%      70.666667
75%      71.666667
max      72.666667
Name: Середня_річна_вологість, dtype: float64


In [19]:
west = df[df["Регіон"] == "Захід"]

print("Регіон Захід:")
print(west)

print("\nОписова статистика:")
print(west["Середня_річна_вологість"].describe())


Регіон Захід:
              Місто Регіон  Січ  Лют  Бер  Кві  Тра  Чер  Лип  Сер  Вер  Жов  \
1             Львів  Захід   85   83   80   73   68   65   63   65   70   77   
6  Івано-Франківськ  Захід   86   84   81   75   70   67   65   67   72   78   
9           Ужгород  Захід   84   82   79   73   68   66   64   66   71   77   

   Лис  Гру  Середня_річна_вологість  
1   82   86                    74.75  
6   83   87                    76.25  
9   82   85                    74.75  

Описова статистика:
count     3.000000
mean     75.250000
std       0.866025
min      74.750000
25%      74.750000
50%      74.750000
75%      75.500000
max      76.250000
Name: Середня_річна_вологість, dtype: float64


In [20]:
east_mean = east["Середня_річна_вологість"].mean()
east_std = east["Середня_річна_вологість"].std()
east_cv = east_std / east_mean * 100

west_mean = west["Середня_річна_вологість"].mean()
west_std = west["Середня_річна_вологість"].std()
west_cv = west_std / west_mean * 100

print("CV Сходу:", east_cv, "%")
print("CV Заходу:", west_cv, "%")


CV Сходу: 2.1514439882421783 %
CV Заходу: 1.1508643239660312 %


In [21]:
# Завдання 4

mean_value = humidity.mean()
median_value = humidity.median()

print("Середнє:", mean_value)
print("Медіана:", median_value)
print("Різниця:", abs(mean_value - median_value))


Середнє: 70.66666666666667
Медіана: 70.5
Різниця: 0.1666666666666714


In [22]:
# Завдання 5

kharkiv_cv = humidity.std() / humidity.mean() * 100

lviv = df[df["Місто"] == "Львів"][months].iloc[0]

lviv_mean = lviv.mean()
lviv_std = lviv.std()
lviv_cv = lviv_std / lviv_mean * 100

print("Харків:")
print("Середнє:", humidity.mean())
print("std:", humidity.std())
print("CV:", kharkiv_cv, "%")

print("\nЛьвів:")
print("Середнє:", lviv_mean)
print("std:", lviv_std)
print("CV:", lviv_cv, "%")


Харків:
Середнє: 70.66666666666667
std: 9.547330263037681
CV: 13.510373013732565 %

Львів:
Середнє: 74.75
std: 8.44366357367787
CV: 11.295871001575748 %


1. У .describe() рядок std відповідає стандартному відхиленню.
Різниця між 75% і 25% дає IQR — міжквартильний розмах.
Різниця між max і min дає розмах.

2. Якщо пропуски заповнювати середнім, нове значення знаходиться
в центрі розподілу. Воно не додає такого розкиду, як реальне
спостереження, тому стандартне відхилення може штучно зменшитися.

3. Сирий std показує абсолютний розкид і залежить від масштабу
значень. Коефіцієнт варіації ділить std на середнє і тому показує
відносний розкид. Це робить його зручнішим для порівняння груп
із різними середніми значеннями.

4. Для Харкова середнє приблизно 69.83%, а медіана 70.5%.
Вони досить близькі, тому розподіл близький до симетричного.
Середнє можна вважати репрезентативною мірою положення для цих даних,
оскільки немає значної різниці між середнім і медіаною.
